source/h2_connection.c

```diff
@@ -1180,7 +1284,15 @@ static void s_handler_installed(struct aws_channel_handler *handler, struct aws_
         }
     }
 
-    if (s_enqueue_settings_frame(connection)) {
+    /* #TODO actually fill with initial settings, now we just disable the push promise for client */
+    /* #TODO Probably we will move the initial settings to connection_new_http2_XXX after we got API for user to change
+     * settings */
+    struct aws_h2_frame_setting initial_settings[2];
+    size_t new_setting_iter = 0;
+    initial_settings[new_setting_iter].id = AWS_H2_SETTINGS_ENABLE_PUSH;
+    initial_settings[new_setting_iter].value = 0;
+    new_setting_iter++;
+    if (aws_h2_connection_change_settings(connection, initial_settings, new_setting_iter)) {
         CONNECTION_LOGF(
             ERROR,
             connection,
```

Comment:
We can remove the s_try_write_outgoing_frames() below this if we move it to aws_h2_connection_change_settings()